src/Ethereum.sol

```diff
@@ -47,65 +36,83 @@ library Ethereum {
     address internal constant WETH_GATEWAY                     = 0xBD7D6a9ad7865463DE44B05F04559f65e3B11704;
 
     /******************************************************************************************************************/
-    /*** Reserve Addresses                                                                                          ***/
+    /*** Reserve Token Addresses                                                                                    ***/
     /******************************************************************************************************************/
 
+    address internal constant DAI                   = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
     address internal constant DAI_ATOKEN            = 0x4DEDf26112B3Ec8eC46e7E31EA5e123490B05B8B;
-    address internal constant DAI_IRM               = 0x7949a8Ef09c49506cCB1cB983317272dcf4170Dd;
-    address internal constant DAI_ORACLE            = 0xAed0c38402a5d19df6E4c03F4E2DceD6e29c1ee9;
     address internal constant DAI_STABLE_DEBT_TOKEN = 0xfe2B7a7F4cC0Fb76f7Fc1C6518D586F1e4559176;
     address internal constant DAI_DEBT_TOKEN        = 0xf705d2B7e92B3F38e6ae7afaDAA2fEE110fE5914;
 
+    address internal constant GNO                   = 0x6810e776880C02933D47DB1b9fc05908e5386b96;
+    address internal constant GNO_ATOKEN            = 0xbf13910620722D4D4F8A03962894EB3335Bf4FaE;
```

Comment:
I think you're pulling the wrong addresses from the pool `getReserveData` - off by one.

This one is `0x7b481aCC9fDADDc9af2cBEA1Ff2342CB1733E50F`.

Reply:
Yep you're right - fixed